src/crawlee/memory_storage_client/_request_queue_client.py

```diff
@@ -200,16 +199,14 @@ async def list_head(self, *, limit: int | None = None) -> RequestQueueHead:
 
                 # Check that the request still exists and was not handled,
                 # in case something deleted it or marked it as handled concurrenctly
-                if request and request.order_no:
-                    requests.append(request)
-
-            items = [request for item in requests if (request := self._json_to_request(item.json_))]
+                if request and not request.handled_at:
+                    requests.append(request.to_request())
```

Comment:
Maybe we can use `internal_request` identifier to distinguish between `Request` and `InternalRequest`?